runelite-client/src/main/java/net/runelite/client/eventbus/EventBus.java

```diff
@@ -53,13 +62,15 @@ private CompositeDisposable getCompositeDisposable(@NonNull Object object)
 			});
 
 		getCompositeDisposable(lifecycle).add(disposable);
+		subscriptionList.put(lifecycle, eventClass);
```

Comment:
Could create a time-of-check/time-of-use race condition but you have to be trying to break things to get that to occur and I don't see any side-effects so it should be alright.